### PR TITLE
Only check flux clusterConfigPath exists when bootstrapping management cluster

### PR DESCRIPTION
**Pull request by jiayiwang7**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/620#" title="Author self-approved">jiayiwang7</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [jiayiwang7]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/addonmanager/addonclients/fluxaddonclient.go [hunk 1]
```diff
@@ -639,20 +639,23 @@ func (fc *fluxForCluster) initializeLocalRepository() error {
...
 }
 
 func (fc *fluxForCluster) validateRemoteConfigPathDoesNotExist(ctx context.Context) error {
-	if exists, err := fc.gitOpts.Git.PathExists(ctx, fc.owner(), fc.repository(), fc.branch(), fc.path()); err != nil {
-		return fmt.Errorf("failed validating remote flux config path: %s", err)
-	} else if exists {
-		return fmt.Errorf("flux path %s already exists in remote repository", fc.path())
+	if fc.clusterSpec.IsSelfManaged() {
+		if exists, err := fc.gitOpts.Git.PathExists(ctx, fc.owner(), fc.repository(), fc.branch(), fc.path()); err != nil {
+			return fmt.Errorf("failed validating remote flux config path: %s", err)
```
Comment (mitalipaygude) [inline pkg/addonmanager/addonclients/fluxaddonclient.go:654]: nit: `%v` instead of `%s`?

**Comment by abhay-krishna** [comment]
New changes are detected. LGTM label has been removed.

File: pkg/addonmanager/addonclients/fluxaddonclient.go (hunk 1 above)
Comment (jiayiwang7) [reply pkg/addonmanager/addonclients/fluxaddonclient.go]: goood catch

**Pull request merged**